In [1]:
import pandas as pd

import warnings
warnings.filterwarnings('ignore')

In [4]:
# Importing the metadata that does not have continents and subregions only countries
metadata_country =pd.read_csv(r"C:\Users\chine\Downloads\Catboost Project\Cleaned Data\Cleaned Data\MetaData.csv" ,encoding ="latin-1")
metadata_country

country_name Country Code                      Region  \
0           Aruba          ABW   Latin America & Caribbean   
1     Afghanistan          AFG                  South Asia   
2          Angola          AGO          Sub-Saharan Africa   
3         Albania          ALB       Europe & Central Asia   
4         Andorra          AND       Europe & Central Asia   
..            ...          ...                         ...   
212        Kosovo          XKX       Europe & Central Asia   
213   Yemen, Rep.          YEM  Middle East & North Africa   
214  South Africa          ZAF          Sub-Saharan Africa   
215        Zambia          ZMB          Sub-Saharan Africa   
216      Zimbabwe          ZWE          Sub-Saharan Africa   

             IncomeGroup                                       SpecialNotes  
0            High income                                                NaN  
1             Low income  The reporting period for national accounts dat...  
2    Lower middle income  The World Bank systematically assesses the app...  
3    Upper middle income                                                NaN  
4            High income                                                NaN  
..                   ...                                                ...  
212  Upper middle income                                                NaN  
213           Low income  The World Bank systematically assesses the app...  
214  Upper middle income  Fiscal year end: March 31; reporting period fo...  
215  Lower middle income  National accounts data were rebased to reflect...  
216  Lower middle income  National Accounts data are reported in Zimbabw...  

[217 rows x 5 columns]

In [5]:
# checking the total number of the countries in metadata
metadata_country['country_name'].nunique()

217

In [3]:
remittance= pd.read_csv(r"C:\Users\chine\Downloads\Catboost Project\Cleaned Data\Cleaned Data\merged_with_data.csv")
remittance

Year country_code country_name       gdp     new_income_group  \
0      1970          AFG  Afghanistan  0.000000           Low income   
1      1971          AFG  Afghanistan  0.000000           Low income   
2      1972          AFG  Afghanistan  0.000000           Low income   
3      1973          AFG  Afghanistan  0.000000           Low income   
4      1974          AFG  Afghanistan  0.000000           Low income   
...     ...          ...          ...       ...                  ...   
14093  2018          ZWE     Zimbabwe  4.179939  Lower middle income   
14094  2019          ZWE     Zimbabwe  6.490458  Lower middle income   
14095  2020          ZWE     Zimbabwe  8.517271  Lower middle income   
14096  2021          ZWE     Zimbabwe  6.985948  Lower middle income   
14097  2022          ZWE     Zimbabwe  9.901347  Lower middle income   

               new_region  population  remittance_growth_rate  \
0              South Asia  10752971.0                0.000000   
1              South Asia  11015857.0                0.000000   
2              South Asia  11286753.0                0.000000   
3              South Asia  11575305.0                0.000000   
4              South Asia  11869879.0                0.000000   
...                   ...         ...                     ...   
14093  Sub-Saharan Africa  15052184.0              -17.468271   
14094  Sub-Saharan Africa  15354608.0               -0.748830   
14095  Sub-Saharan Africa  15669666.0               29.288915   
14096  Sub-Saharan Africa  15993524.0                8.185447   
14097  Sub-Saharan Africa  16320537.0                3.300000   

       remittance_paid  remittance_per_capita  remittance_received  \
0                 0.00               0.000000         0.000000e+00   
1                 0.00               0.000000         0.000000e+00   
2                 0.00               0.000000         0.000000e+00   
3                 0.00               0.000000         0.000000e+00   
4                 0.00               0.000000         0.000000e+00   
...                ...                    ...                  ...   
14093      29278892.45              94.850224         1.427703e+09   
14094      13480565.66              92.285779         1.417012e+09   
14095      21707094.03             116.916301         1.832039e+09   
14096             0.00             123.925159         1.982000e+09   
14097             0.00             125.449671         2.047406e+09   

       remittance_to_gdp_ratio  remittance_volatility  unemployment_rate  \
0                 0.000000e+00           2.934855e+08              0.000   
1                 0.000000e+00           2.934855e+08              0.000   
2                 0.000000e+00           2.934855e+08              0.000   
3                 0.000000e+00           2.934855e+08              0.000   
4                 0.000000e+00           2.934855e+08              0.000   
...                        ...                    ...                ...   
14093             3.415607e+08           8.881924e+08              6.784   
14094             2.183223e+08           8.881924e+08              7.370   
14095             2.150970e+08           8.881924e+08              7.898   
14096             2.837124e+08           8.881924e+08              8.067   
14097             2.067806e+08           8.881924e+08              7.945   

       unemployment_rate_change  
0                         0.000  
1                         0.000  
2                         0.000  
3                         0.000  
4                         0.000  
...                         ...  
14093                     0.505  
14094                     0.586  
14095                     0.528  
14096                     0.169  
14097                    -0.122  

[14098 rows x 15 columns]

In [6]:
# checking the total number of the remittance data
remittance['country_name'].nunique()

266

There are 217 countries in the metadata  But we have 266  in the remittance data set
We need to find the countries that are not supposed to be in the remittance data

In [7]:
# Getting the names of the countries in the remittance datafarme and storing them in a list
remittance_country_list = remittance['country_name'].unique()
remittance_country_list

array(['Afghanistan', 'Africa Eastern and Southern',
       'Africa Western and Central', 'Albania', 'Algeria',
       'American Samoa', 'Andorra', 'Angola', 'Antigua and Barbuda',
       'Arab World', 'Argentina', 'Armenia', 'Aruba', 'Australia',
       'Austria', 'Azerbaijan', 'Bahamas, The', 'Bahrain', 'Bangladesh',
       'Barbados', 'Belarus', 'Belgium', 'Belize', 'Benin', 'Bermuda',
       'Bhutan', 'Bolivia', 'Bosnia and Herzegovina', 'Botswana',
       'Brazil', 'British Virgin Islands', 'Brunei Darussalam',
       'Bulgaria', 'Burkina Faso', 'Burundi', 'Cabo Verde', 'Cambodia',
       'Cameroon', 'Canada', 'Caribbean small states', 'Cayman Islands',
       'Central African Republic', 'Central Europe and the Baltics',
       'Chad', 'Channel Islands', 'Chile', 'China', 'Colombia', 'Comoros',
       'Congo, Dem. Rep.', 'Congo, Rep.', 'Costa Rica', "Cote d'Ivoire",
       'Croatia', 'Cuba', 'Curacao', 'Cyprus', 'Czechia', 'Denmark',
       'Djibouti', 'Dominica', 'Dominican Repub

In [10]:
len(remittance_country_list)

266

In [8]:
# Lets get the unique names of countries in metadata  and Create a list to store the  countries 
metadata_country_list = metadata_country['country_name'].unique() 
metadata_country_list

array(['Aruba', 'Afghanistan', 'Angola', 'Albania', 'Andorra',
       'United Arab Emirates', 'Argentina', 'Armenia', 'American Samoa',
       'Antigua and Barbuda', 'Australia', 'Austria', 'Azerbaijan',
       'Burundi', 'Belgium', 'Benin', 'Burkina Faso', 'Bangladesh',
       'Bulgaria', 'Bahrain', 'Bahamas, The', 'Bosnia and Herzegovina',
       'Belarus', 'Belize', 'Bermuda', 'Bolivia', 'Brazil', 'Barbados',
       'Brunei Darussalam', 'Bhutan', 'Botswana',
       'Central African Republic', 'Canada', 'Switzerland',
       'Channel Islands', 'Chile', 'China', "Cote d'Ivoire", 'Cameroon',
       'Congo, Dem. Rep.', 'Congo, Rep.', 'Colombia', 'Comoros',
       'Cabo Verde', 'Costa Rica', 'Cuba', 'Curacao', 'Cayman Islands',
       'Cyprus', 'Czechia', 'Germany', 'Djibouti', 'Dominica', 'Denmark',
       'Dominican Republic', 'Algeria', 'Ecuador', 'Egypt, Arab Rep.',
       'Eritrea', 'Spain', 'Estonia', 'Ethiopia', 'Finland', 'Fiji',
       'France', 'Faroe Islands', 'Micronesia, Fed

In [9]:
len(metadata_country_list)

217

In [13]:
# Lets confirm if the number of coutries we have in our remittance dataset is equal to the number of countries
# we have metadata set
remittance_country_list == metadata_country_list

False

In [14]:
# a dataframe to extract only the countries in metadata country dataset

realcountries = remittance[remittance['country_name'].isin(metadata_country_list)]
realcountries

Year country_code country_name       gdp     new_income_group  \
0      1970          AFG  Afghanistan  0.000000           Low income   
1      1971          AFG  Afghanistan  0.000000           Low income   
2      1972          AFG  Afghanistan  0.000000           Low income   
3      1973          AFG  Afghanistan  0.000000           Low income   
4      1974          AFG  Afghanistan  0.000000           Low income   
...     ...          ...          ...       ...                  ...   
14093  2018          ZWE     Zimbabwe  4.179939  Lower middle income   
14094  2019          ZWE     Zimbabwe  6.490458  Lower middle income   
14095  2020          ZWE     Zimbabwe  8.517271  Lower middle income   
14096  2021          ZWE     Zimbabwe  6.985948  Lower middle income   
14097  2022          ZWE     Zimbabwe  9.901347  Lower middle income   

               new_region  population  remittance_growth_rate  \
0              South Asia  10752971.0                0.000000   
1              South Asia  11015857.0                0.000000   
2              South Asia  11286753.0                0.000000   
3              South Asia  11575305.0                0.000000   
4              South Asia  11869879.0                0.000000   
...                   ...         ...                     ...   
14093  Sub-Saharan Africa  15052184.0              -17.468271   
14094  Sub-Saharan Africa  15354608.0               -0.748830   
14095  Sub-Saharan Africa  15669666.0               29.288915   
14096  Sub-Saharan Africa  15993524.0                8.185447   
14097  Sub-Saharan Africa  16320537.0                3.300000   

       remittance_paid  remittance_per_capita  remittance_received  \
0                 0.00               0.000000         0.000000e+00   
1                 0.00               0.000000         0.000000e+00   
2                 0.00               0.000000         0.000000e+00   
3                 0.00               0.000000         0.000000e+00   
4                 0.00               0.000000         0.000000e+00   
...                ...                    ...                  ...   
14093      29278892.45              94.850224         1.427703e+09   
14094      13480565.66              92.285779         1.417012e+09   
14095      21707094.03             116.916301         1.832039e+09   
14096             0.00             123.925159         1.982000e+09   
14097             0.00             125.449671         2.047406e+09   

       remittance_to_gdp_ratio  remittance_volatility  unemployment_rate  \
0                 0.000000e+00           2.934855e+08              0.000   
1                 0.000000e+00           2.934855e+08              0.000   
2                 0.000000e+00           2.934855e+08              0.000   
3                 0.000000e+00           2.934855e+08              0.000   
4                 0.000000e+00           2.934855e+08              0.000   
...                        ...                    ...                ...   
14093             3.415607e+08           8.881924e+08              6.784   
14094             2.183223e+08           8.881924e+08              7.370   
14095             2.150970e+08           8.881924e+08              7.898   
14096             2.837124e+08           8.881924e+08              8.067   
14097             2.067806e+08           8.881924e+08              7.945   

       unemployment_rate_change  
0                         0.000  
1                         0.000  
2                         0.000  
3                         0.000  
4                         0.000  
...                         ...  
14093                     0.505  
14094                     0.586  
14095                     0.528  
14096                     0.169  
14097                    -0.122  

[11501 rows x 15 columns]

In [12]:
# checking the total number of the remmitance data
realcountries['country_name'].nunique()

217

In [15]:
# Getting the names of the countries that are not metadata country data set
print(set(remittance_country_list) ^ set(metadata_country_list))

{'Middle East & North Africa (IDA & IBRD countries)', 'IBRD only', 'IDA total', 'Europe & Central Asia (excluding high income)', 'Not classified', 'Latin America & Caribbean (excluding high income)', 'Small states', 'East Asia & Pacific (excluding high income)', 'South Asia', 'Caribbean small states', 'Arab World', 'IDA blend', 'East Asia & Pacific', 'World', 'Post-demographic dividend', 'Lower middle income', 'Africa Western and Central', 'Central Europe and the Baltics', 'Middle East & North Africa (excluding high income)', 'Pre-demographic dividend', 'Europe & Central Asia', 'Low & middle income', 'Sub-Saharan Africa (excluding high income)', 'Low income', 'European Union', 'Least developed countries: UN classification', 'Heavily indebted poor countries (HIPC)', 'South Asia (IDA & IBRD)', 'Sub-Saharan Africa (IDA & IBRD countries)', 'Middle East & North Africa', 'Europe & Central Asia (IDA & IBRD countries)', 'Africa Eastern and Southern', 'IDA only', 'Pacific island small states', 

In [18]:
#Renaming the column names for better understanding
realcountries.rename(columns={'new_income_group': 'income_group', 'new_region': 'region'}, inplace=True)

In [19]:
realcountries.to_csv(r"C:\Users\chine\Downloads\Catboost Project\Cleaned Data\Remittance 1970 - 2022 -Chinelo.csv", index = False)   #To save to csv